lib/rpc/GlobalRpcProviders.ts

```diff
@@ -2,87 +2,96 @@ import { ChainId } from '@uniswap/sdk-core'
 import { SingleJsonRpcProvider } from './SingleJsonRpcProvider'
 import { UniJsonRpcProvider } from './UniJsonRpcProvider'
 import Logger from 'bunyan'
-import { SUPPORTED_CHAINS } from '@uniswap/smart-order-router'
 import {
   DEFAULT_SINGLE_PROVIDER_CONFIG,
   DEFAULT_UNI_PROVIDER_CONFIG,
   SingleJsonRpcProviderConfig,
   UniJsonRpcProviderConfig,
 } from './config'
+import { ProdConfig, ProdConfigJoi } from './ProdConfig'
+import { chainIdToNetworkName } from './utils'
 
 export class GlobalRpcProviders {
-  private static readonly PROVIDER_RPC_URL_RANKING: Map<ChainId, number[] | undefined> = new Map([
-    [ChainId.AVALANCHE, undefined],
-  ])
-
-  private static readonly PROVIDER_RPC_URL_WEIGHTS: Map<ChainId, number[] | undefined> = new Map([
-    [ChainId.AVALANCHE, undefined],
-  ])
-
   private static SINGLE_RPC_PROVIDERS: Map<ChainId, SingleJsonRpcProvider[]> | null = null
 
   private static UNI_RPC_PROVIDERS: Map<ChainId, UniJsonRpcProvider> | null = null
 
-  private static initGlobalSingleRpcProviders(log: Logger, config: SingleJsonRpcProviderConfig) {
-    // Only Avalanche is supported for now.
-    const infuraAvalancheUrl = process.env[`WEB3_RPC_${ChainId.AVALANCHE.toString()}`]!
-    if (infuraAvalancheUrl === undefined) {
+  private static getProdConfig(): ProdConfig {
+    const prodConfigStr = process.env['UNI_RPC_PROVIDER_PROD_CONFIG']!
+    if (prodConfigStr === undefined) {
+      throw new Error('Environment variable UNI_RPC_PROVIDER_PROD_CONFIG is missing!')
+    }
+    const validation = ProdConfigJoi.validate(JSON.parse(prodConfigStr))
```

Comment:
do we want to fail or do we want to fallback to a default wdyt? if it fails then literally our routing-api goes down right?